### DISCUSSION TITLE
docs/v1.12.x/guide/vpc/

### DISCUSSION THREAD

**Discussion by giscus** [kubeovn/kube-ovn#2644] [Announcements]
# docs/v1.12.x/guide/vpc/

https://kubeovn.github.io/docs/v1.12.x/guide/vpc/

**Comment by zbb88888** [comment]
在使用vpc-nat-gw时需要注意：
1. nat gw创建之后，net1 arp时关闭的，ping不通 网关，创建eip后会自动打开arp，可以ping通。
2. Hyper-V及其他虚拟机下的网络无法直接使用macvlan连通Node网络。

**Reply by oilbeater** [reply]
https://serverfault.com/questions/812528/docker-container-connection-in-macvlan-network 看上去 hyper-v 更改网络选项可以支持 macvlan

**Comment by WaterAndBread** [comment]
配置 vpc-dns 的 ConfigMap   时， enable-vpc-dns: true   需为 enable-vpc-dns: “true” ， 

不然会报 Error from server (BadRequest): error when creating "vpc-dns": ConfigMap in version "v1" cannot be handled as a ConfigMap: json: cannot unmarshal bool into Go struct field ConfigMap.data of type string

**Reply by oilbeater** [reply]
已修复，多谢

**Comment by aliniun** [comment]
vpn-dns部署的前提是需要有coredns吗？

**Reply by aliniun** [reply]
func (c *Controller) getDefaultCoreDNSImage() (string, error) {
	dp, err := c.config.KubeClient.AppsV1().Deployments("kube-system").
		Get(context.Background(), "coredns", metav1.GetOptions{})
	if err != nil {
		klog.Error(err)
		return "", err
	}

	for _, container := range dp.Spec.Template.Spec.Containers {
		if container.Name == CorednsContainerName {
			return container.Image, nil
		}
	}

	return "", fmt.Errorf("coredns container no found")
}

**Reply by zbb88888** [reply]
这个coredns 镜像是需要的，文档里面应该有写把这个镜像配置到一个 comfigmap里面

**Reply by aliniun** [reply]
感谢回复，其实我指的是是否需要有coredns这个pod存在？

**Reply by zbb88888** [reply]
coredns 应该是要有的，而且目前部署 k8s的框架， coredns 一般都是默认的 dns 组件。

**Comment by kldancer** [comment]
请问一下，存在多个vpc的情况下，如果创建vpc及子网时没有指定namespace，那创建pod时如何指定使用特定的自定义vpc呢？@bobz965

**Reply by zbb88888** [reply]
ovn 的子网都属于一个vpc，创建 pod 的时候基于 annos 指定子网应该就可以
